ral-facilities/operationsgateway
# DSEGOG-17 adding Redux state management

## Description
To allow for complex state management, I am adding Redux. I have based a lot of what I've done on how DataGateway implements Redux, intending to correctly segregate SciGateway state and OperationsGateway state. This also allows for us to start making use of an OperationsGateway settings file, and I've done so by adding in an apiUrl, routes array and pluginHost (none of which are really being used at the moment but will be later on). 

It's likely that, in my attempt to mimic Redux in DataGateway's monorepo structure, I have added something unnecessary or missed something out that we need. Louise will know more about this and can help here.

## Testing instructions
Look over the code and see how things look. I tested that an apiUrl was being read from the settings file by console.logging it in the useRecordsPaginated function, where it is fetched for future use. I've done my best to add tests that ensure the correct actions are firing to store info in state but always good to check this yourself.


## Agile board tracking
Closes DSEGOG-17
